src/trunk.jl

```diff
@@ -129,6 +129,20 @@ end
   return solve!(solver, nlp; x = x, kwargs...)
 end
 
+"""
+    normM!(n, x, M, z)
+    
+Weighted norm of `x` with respect to `M`, i.e., `z = sqrt(x' * M * x)`. Uses `z` as workspace.
+"""
+function normM!(x, M, z)
+  if M === I
+    return norm(x)
```

Comment:
```suggestion
    return nrm2(n, x)
```
The package use `nrm2`, so I would prefer to keep it this way in the PR.